### PR TITLE
Display whole path in the `Site or Folder` dropdown

### LINKED ISSUE #23681: Let's show the whole path when select a folder in the host selector
[](https://mrkr.io/s/63ac6e78f92bad90de682101/0)

---

**Reported by:** Melissa Rojas Rodríguez (melissa.rojas@dotcms.com)

**Source URL:** [https://demo.dotcms.com/dotAdmin/#/c/c_Blog-Entries](https://demo.dotcms.com/dotAdmin/#/c/c_Blog-Entries)

**Issue details:** [Open in Marker.io](https://app.marker.io/i/63ac6e78f92bad90de682104_ea4aeec31b13723b?advanced=1)

<table><tr><td><strong>Device type</strong></td><td>desktop</td></tr><tr><td><strong>Browser</strong></td><td>Chrome 108.0.0.0</td></tr><tr><td><strong>Screen Size</strong></td><td>1440 x 900</td></tr><tr><td><strong>OS</strong></td><td>OS X 10.14.6</td></tr><tr><td><strong>Viewport Size</strong></td><td>1440 x 821</td></tr><tr><td><strong>Zoom Level</strong></td><td>100%</td></tr><tr><td><strong>Pixel Ratio</strong></td><td>@&#8203;2x</td></tr></table>

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* We're now displaying the whole path of the selected site/folder in the `Site or Folder` drop-downs. This way, Users can see the actual path of the folder that has been selected.

### CODE REVIEW

**Comment by zulqarnainvd** [comment]
[screen-capture (23).webm](https://user-images.githubusercontent.com/113915849/227920897-bedd7a63-9801-4492-8dc8-a5c523b061e3.webm)

**Comment by zulqarnainvd** [comment]
[screen-capture (24).webm](https://user-images.githubusercontent.com/113915849/227921672-98ad125e-f6df-458d-9b67-a3605c862d01.webm)


### Old

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/tree/HostFolderTreeReadStoreModel.js [hunk 1]
```diff
@@ -115,6 +115,17 @@ dojo.declare("dotcms.dijit.tree.HostFolderTreeReadStoreModel", null, {
...
 
+	getFullPath: function(item) {
+		let fullPath = "";
+		const Host = item.fullPath.split(":")[0];
+		if (item.type === "host") {
+			fullPath = Host
+		} else {
+			fullPath = `${Host}${item.folderPath}`
+		}
+		return fullPath
```
Comment (zJaaal) [inline dotCMS/src/main/webapp/html/js/dotcms/dijit/tree/HostFolderTreeReadStoreModel.js:126]: You can remove the if condition and the fullPath variable by using a ternary operator and you can use destructuring to take the first index of an array
```suggestion
                const [ host ] = item.fullPath.split(":");
		return item.type === "host" ? Host : `${Host}${item.folderPath}`
```
Also I recommend to change `Host` for `host` to maintain the standard

**Review by hmoreras — APPROVED** [review]

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/tree/HostFolderTreeReadStoreModel.js (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/webapp/html/js/dotcms/dijit/tree/HostFolderTreeReadStoreModel.js]: Done!

**Review by jdotcms — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Pull request merged**